xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-ui/src/main/resources/XWiki/Lightbox/GetPageAttachmentsService.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.5" reference="XWiki.Lightbox.GetPageAttachmentsService" locale="">
+  <web>XWiki.Lightbox</web>
+  <name>GetPageAttachmentsService</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>XWiki.Lightbox.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <originalMetadataAuthor>xwiki:XWiki.Admin</originalMetadataAuthor>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>GetPageAttachmentsService</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+#if ($request.imageURL)
+  #set ($attachReference = $services.model.resolveAttachment($request.imageURL, 'resource/standard'))
+  #set ($attachDocument = $xwiki.getDocument($attachReference.getDocumentReference()))
```

Comment:
I think it might be a good idea to output an error if the document is `null`, which is the case if the user doesn't have access to it.

Reply:
Okay, maybe also not, I guess this is catched by the final if due to Velocity's ignoring of undefined variables?

Reply:
I agree with @michitux that it would be better to have an explicit test or at least an explanation to explain where it's checked and why it's working ;)